tests/unit_test/test_ga_pipeline.py

```diff
@@ -0,0 +1,165 @@
+    with patch.object(
+            ga_pipeline_module, 'get_current_timestamp_as_string'
+    ) as mock:
+        yield mock
+
+
+GA_CONFIG = {
+    'pipelineID': 'pipeline_id',
+    'dataset': 'dataset',
+    'table': 'table',
+    'viewId': 'viewId',
+    'dimensions': ['dimension'],
+    'metrics': ['metrics'],
+    'recordAnnotations': [{
+        'recordAnnotationFieldName': 'recordAnnotationFieldName',
+        'recordAnnotationValue': 'recordAnnotationValue'
+    }]
+}
+
+
+TEST_DATA = {
+        'reports':
+        [
+            {
+                'data':
+                    {
+                        'rows': [
+                            {
+                                'metrics': [
+                                    {'values': ['12566', '1172250.0']}
+                                ],
+                                'dimensions': ['New Visitor']
+                            },
+                            {
+                                'metrics': [
+                                    {'values': ['12798', '2215033.0']}
+                                ],
+                                'dimensions': ['Returning Visitor']
+                            }
+                        ],
+                        'maximums': [
+                            {'values': ['12798', '2215033.0']}
+                        ],
+                        'minimums': [
+                            {'values': ['12566', '1172250.0']}
+                        ],
+                        'isDataGolden': True,
+                        'totals': [{'values': ['25364', '3387283.0']}],
+                        'rowCount': 2
+                    },
+                'columnHeader': {
+                    'dimensions': ['ga:userType'],
+                    'metricHeader': {
+                        'metricHeaderEntries': [
+                            {'type': 'INTEGER', 'name': 'ga:sessions'},
+                            {'type': 'TIME',
+                             'name': 'ga:sessionDuration'}
+                        ]
+                    }
+                    }
+
+            }
+        ]
+    }
+
+
+class TestTransformResponseToBQCompatibleRecords:
+    def test_should_return_empty_list_when_input_is_empty(self):
+        resp = list(transform_response_to_bq_compatible_record({}))
+        assert len(resp) == 0
+
+    def test_should_transform_response_to_bq_compatible_record(self):
+
+        response = list(transform_response_to_bq_compatible_record(TEST_DATA))
+
+        expected_data = [
+            {'ga_userType': 'New Visitor', 'date_range': '0',
+             'ga_sessions': '12566', 'ga_sessionDuration': '1172250.0'},
+            {'ga_userType': 'Returning Visitor', 'date_range': '0',
+             'ga_sessions': '12798', 'ga_sessionDuration': '2215033.0'}
+        ]
+
+        assert response == expected_data
+
+
+class TestETLGA:
+    # pylint: disable=too-many-arguments
+    def test_should_etl_ga(
```

Comment:
minor: naming is hard. I guess this just tests that it is not failing and calls the right methods (without checking what is passed).

Reply:
yeah